sycl/plugins/unified_runtime/adapters/level_zero/ur_level_zero.hpp

```diff
@@ -0,0 +1,217 @@
+      {ZE_RESULT_ERROR_UNSUPPORTED_ALIGNMENT, ZER_RESULT_INVALID_VALUE},
+      {ZE_RESULT_ERROR_INVALID_SYNCHRONIZATION_OBJECT,
+       ZER_RESULT_INVALID_EVENT},
+      {ZE_RESULT_ERROR_INVALID_ENUMERATION, ZER_RESULT_INVALID_VALUE},
+      {ZE_RESULT_ERROR_UNSUPPORTED_ENUMERATION, ZER_RESULT_INVALID_VALUE},
+      {ZE_RESULT_ERROR_UNSUPPORTED_IMAGE_FORMAT, ZER_RESULT_INVALID_VALUE},
+      {ZE_RESULT_ERROR_INVALID_NATIVE_BINARY, ZER_RESULT_INVALID_BINARY},
+      {ZE_RESULT_ERROR_INVALID_KERNEL_NAME, ZER_RESULT_INVALID_KERNEL_NAME},
+      {ZE_RESULT_ERROR_INVALID_FUNCTION_NAME,
+       ZER_RESULT_ERROR_INVALID_FUNCTION_NAME},
+      {ZE_RESULT_ERROR_OVERLAPPING_REGIONS, ZER_RESULT_INVALID_OPERATION},
+      {ZE_RESULT_ERROR_INVALID_GROUP_SIZE_DIMENSION,
+       ZER_RESULT_INVALID_WORK_GROUP_SIZE},
+      {ZE_RESULT_ERROR_MODULE_BUILD_FAILURE,
+       ZER_RESULT_ERROR_MODULE_BUILD_FAILURE},
+      {ZE_RESULT_ERROR_OUT_OF_DEVICE_MEMORY,
+       ZER_RESULT_ERROR_OUT_OF_DEVICE_MEMORY},
+      {ZE_RESULT_ERROR_OUT_OF_HOST_MEMORY,
+       ZER_RESULT_ERROR_OUT_OF_HOST_MEMORY}};
+
+  auto It = ErrorMapping.find(ZeResult);
+  if (It == ErrorMapping.end()) {
+    return ZER_RESULT_ERROR_UNKNOWN;
+  }
+  return It->second;
+}
+
+// Controls Level Zero calls tracing.
+enum DebugLevel {
+  ZE_DEBUG_NONE = 0x0,
+  ZE_DEBUG_BASIC = 0x1,
+  ZE_DEBUG_VALIDATION = 0x2,
+  ZE_DEBUG_CALL_COUNT = 0x4,
+  ZE_DEBUG_ALL = -1
+};
+
+const int ZeDebug = [] {
+  const char *DebugMode = std::getenv("ZE_DEBUG");
+  return DebugMode ? std::atoi(DebugMode) : ZE_DEBUG_NONE;
+}();
+
+// Prints to stderr if ZE_DEBUG allows it
+void zePrint(const char *Format, ...);
+
+// This function will ensure compatibility with both Linux and Windows for
+// setting environment variables.
+bool setEnvVar(const char *name, const char *value);
+
+// Perform traced call to L0 without checking for errors
+#define ZE_CALL_NOCHECK(ZeName, ZeArgs)                                        \
+  ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs, false)
+
+// Flags which tell whether various Level Zero extensions are available.
+static bool ZeDriverGlobalOffsetExtensionFound = false;
+static bool ZeDriverModuleProgramExtensionFound = false;
+
+struct _ur_level_zero_platform : public _ur_platform {
+  _ur_level_zero_platform(ze_driver_handle_t Driver) : ZeDriver{Driver} {}
+  // Performs initialization of a newly constructed PI platform.
+  zer_result_t initialize();
+
+  // Level Zero lacks the notion of a platform, but there is a driver, which is
+  // a pretty good fit to keep here.
+  ze_driver_handle_t ZeDriver;
+
+  // Cache versions info from zeDriverGetProperties.
+  std::string ZeDriverVersion;
+  std::string ZeDriverApiVersion;
+  ze_api_version_t ZeApiVersion;
+
+  // Cache driver extensions
+  std::unordered_map<std::string, uint32_t> zeDriverExtensionMap;
+};
+
+using ur_level_zero_platform = _ur_level_zero_platform *;
+
+class ZeUSMImportExtension {

```

Comment:
minor: since now these classes are defined in the UR, should they be named Zer? if so, we could do it in a follow-up patch.

Reply:
I think everything that goes into "unified_runtime/" should now be named ZER (actually renamed to UR at some point), but anything that goes into "unified_runtime/adapters/level_zero" can use ZE in their names.